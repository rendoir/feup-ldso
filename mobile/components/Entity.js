import React from 'react';
import {
    StyleSheet
} from 'react-native';

import { Text, ListItem, Button } from 'native-base';

export default class Entity extends React.Component {

    constructor(props) {
        super(props);

    }

    render() {
        return (
            <ListItem style={styles.listItem} onPress={this.props.onPress} >
                <Button transparent onPress={this.props.onPress}><Text style={styles.buttonText}>{this.props.data.initials}</Text></Button>
            </ListItem>
        );
    }
}

const styles = StyleSheet.create({
    buttonText: {
        color: 'white',
        fontFamily: 'OpenSans-Regular',
        fontSize: 18,
        paddingLeft: 5
    },
    listItem: {
        justifyContent: "center",
        backgroundColor: '#7C8589',
        width: '91%',
        borderBottomWidth: 1,
        borderBottomColor: 'white'
    }
});
