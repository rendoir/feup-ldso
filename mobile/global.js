global.api = "188.166.95.23";
global.userId = null;
global.userName = null;

global.dictionary = {
    FAVORITES: { PT: "Favoritos", EN: "Favorites" },
    AGENDA: { PT: "Agenda", EN: "Agenda" },
    SEARCH: { PT: "Pesquisa", EN: "Search" },
    EVENTS: { PT: "Eventos", EN: "Events" },
    CATEGORY: { PT: "Categoria", EN: "Category" },
    CATEGORIES: { PT: "Categorias", EN: "Categories" },
    ENTITY: { PT: "Entidade", EN: "Entity" },
    ENTITIES: { PT: "Entidades", EN: "Entities" },
    ALL: { PT: "Todos", EN: "All" },
    NO_EVENTS: { PT: "Não há eventos futuros para mostrar.", EN: "There are no future events to display." },
    AT: { PT: "em", EN: "at" },
    PRICE: { PT: "Preço", EN: "Price" },
    FREE: { PT: "Gratuito", EN: "Free" },
    MAP: { PT: "Mapa", EN: "Map" },
    CALENDAR: { PT: "Calendário", EN: "Calendar" },
    APP_NAME: { PT: "Agenda Cultural da Universidade do Porto", EN: "Porto University's Cultural Agenda" },
    GOOGLE_SIGNIN: { PT: "Entrar com conta Google", EN: "Sign in with Google account" },
    NO_SEARCH_RESULTS: { PT: "Não há eventos para a pesquisa feita.", EN: "There are no events that match your search." },
    NO_FAVORITES: { PT: "Não há eventos favoritos para mostrar.", EN: "There are no favorite events." }
};

global.calendar = {
    1: { PT: 'JAN', EN: 'JAN' },
    2: { PT: 'FEV', EN: 'FEB' },
    3: { PT: 'MAR', EN: 'MAR' },
    4: { PT: 'ABR', EN: 'APR' },
    5: { PT: 'MAI', EN: 'MAY' },
    6: { PT: 'JUN', EN: 'JUN' },
    7: { PT: 'JUL', EN: 'JUL' },
    8: { PT: 'AGO', EN: 'AUG' },
    9: { PT: 'SET', EN: 'SEP' },
    10: { PT: 'OUT', EN: 'OCT' },
    11: { PT: 'NOV', EN: 'NOV' },
    12: { PT: 'DEZ', EN: 'DEC' }
};

global.fullCalendar = {
    1: { PT: 'Janeiro', EN: 'January' },
    2: { PT: 'Fevereiro', EN: 'February' },
    3: { PT: 'Março', EN: 'March' },
    4: { PT: 'Abril', EN: 'April' },
    5: { PT: 'Maio', EN: 'May' },
    6: { PT: 'Junho', EN: 'June' },
    7: { PT: 'Julho', EN: 'July' },
    8: { PT: 'Agosto', EN: 'August' },
    9: { PT: 'Setembro', EN: 'September' },
    10: { PT: 'Outubro', EN: 'October' },
    11: { PT: 'Novembro', EN: 'November' },
    12: { PT: 'Dezembro', EN: 'December' }
};
