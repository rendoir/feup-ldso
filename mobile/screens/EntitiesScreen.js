import React from 'react';
import { StyleSheet } from 'react-native';
import { Font, AppLoading } from "expo";
import { Root, Container, Header, List, Text, View, Content, ListItem, Button } from 'native-base';
import axios from 'axios';
import Entity from '../components/Entity';

export default class EntitiesScreen extends React.Component {
  state = {
    loading: true,
    entities: []
  };
  static navigationOptions = {
    header: null,
  };

  componentWillUnmount() {
    this.isCancelled = true;
  }

  async componentWillMount() {
    await Font.loadAsync({
      Roboto: require("native-base/Fonts/Roboto.ttf"),
      Roboto_medium: require("native-base/Fonts/Roboto_medium.ttf")
    });
    this.getEntitiesFromApi();
    !this.isCancelled && this.setState({ loading: false });
  }


  getEntitiesFromApi() {
    let self = this;
    axios.get('http://' + global.api + ':3030/app/entities')
      .then(function (response) {
        const entities = response.data;
        self.setState({ entities });
      })
      .catch(function (error) {
        console.log(error);
      });
  }

  render() {
    if (this.state.loading) {
      return (
        <Root>
          <AppLoading />
        </Root>
      );
    }
    const { navigate } = this.props.navigation;

    const entities = this.state.entities.map((entity, i) => (
      <Entity data={entity} key={i} onPress={() => navigate('Agenda', { selectedEntity: entity.initials, selectedEntityId: entity.id })} />
    ));

    return (
      <Container>
        <Header style={styles.header}>
          <Text style={styles.headerText}>Organizações</Text>
        </Header>
        <Content>
          <List>
            <ListItem style={styles.listItem} onPress={() => navigate('Agenda', { selectedEntity: 'Orgão', selectedEntityId: 'null' })}>
              <Button transparent onPress={() => navigate('Agenda', { selectedEntity: 'Orgão', selectedEntityId: 'null' })}><Text style={styles.buttonText}>Todos</Text></Button>
            </ListItem>
            {entities}
          </List>
        </Content>
      </Container>
    )
  }
}

const styles = StyleSheet.create({
  listItem: {
    justifyContent: "center",
  },
  header: {
    backgroundColor: '#2c8f7f',
    height: 80,
  },
  headerText: {
    color: 'white',
    fontSize: 30,
    textAlign: 'center',
    width: '100%',
    alignSelf: 'center',
  },
  buttonText: {
    color: 'black',
    fontFamily: 'OpenSans-Regular',
    fontSize: 18,
    paddingLeft: 5,
  },
});