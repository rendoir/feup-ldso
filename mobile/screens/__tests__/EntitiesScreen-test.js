/**
 * @jest-environment jsdom
 */
import 'react-native';
import React from 'react';
import EntitiesScreen from '../EntitiesScreen';
import Entity from '../../components/Entity';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '../../global.js';

Enzyme.configure({ adapter: new Adapter() });
var mockAxios = new MockAdapter(axios);

let entities = [
    {
        id: 1,
        initials: "FEUP"
    },
    {
        id: 2,
        initials: "FMUP"
    },
    {
        id: 3,
        initials: "FLUP"
    }
];

describe('App snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders correctly', async() => {
        const tree = renderer.create(<EntitiesScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Get Entities', () => {

    it('get entities from api and saves them', async() => {

        mockAxios.onGet().reply(200, entities);

        let wrapper = await shallow(<EntitiesScreen screenProps={{ language: 'PT' }} />);

        await wrapper.instance().getEntitiesFromApi();

        setImmediate(() => {
            expect(wrapper.state().entities.length).toEqual(3);
        });

    });

});

describe('Get Entities Navigation', () => {

    it('entity press navigation', async() => {

        mockAxios.onGet().reply(200, entities);

        const navigate = jest.fn();
        const navigation = { navigate: navigate, state: { params: { onSelect: jest.fn() } } };
        let wrapper = await shallow(<EntitiesScreen screenProps={{ language: 'PT' }} navigation={navigation} />);

        wrapper.setState({ loading: false, entities: entities });

        let entity = wrapper.find(Entity).first();
        entity.props().onPress();

        expect(navigate).toHaveBeenCalled();

    });

    it('list item press navigation', async() => {

        mockAxios.onGet().reply(200, entities);

        const navigate = jest.fn();
        const navigation = { navigate: navigate, state: { params: { onSelect: jest.fn() } } };
        let wrapper = await shallow(<EntitiesScreen screenProps={{ language: 'PT' }} navigation={navigation} />);
        wrapper.setState({ loading: false, entities: entities });

        let list = wrapper.find(".list-entities").first();
        list.props().onPress();

        expect(navigate).toHaveBeenCalled();

        wrapper.unmount();

    });

    it('button all categories navigation', async() => {

        mockAxios.onGet().reply(200, entities);

        const navigate = jest.fn();
        const navigation = { navigate: navigate, state: { params: { onSelect: jest.fn() } } };
        let wrapper = await shallow(<EntitiesScreen screenProps={{ language: 'PT' }} navigation={navigation} />);
        wrapper.setState({ loading: false, entities: entities });

        let button = wrapper.find(".all-entities").first();
        button.props().onPress();

        expect(navigate).toHaveBeenCalled();

    });


});
