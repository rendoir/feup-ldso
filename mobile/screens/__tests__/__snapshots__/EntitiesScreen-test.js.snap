// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "maxHeight": 667,
            "minHeight": 56,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      />
    </View>
  </Modal>
</View>
`;
